### PR TITLE
Add support for metadata hash verification.

### PR DESCRIPTION
**Pull request by gianfra-t**
Context: [Support](https://forum.polkadot.network/t/new-polkadot-ledger-app/8817#adding-support-to-your-parachain-runtime-8) for new Ledger App.

Closes [No ticket related, small change].

Following this [guide](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/guides/enable_metadata_hash/index.html), here we add the relevant code for the runtime to accept the verification of the metadata that was used for signing the extrinsic.

Note: the command to build the node for production now is `cargo build --release --features metadata-hash`.

It is important to note given that is not clear on the guide, that the environment variable required when building with this new feature (`RUNTIME_METADATA_HASH`) is set by the build script when the flag `metadata-hash` is used. See [here](https://github.com/paritytech/polkadot-sdk/blob/b9eb68bcb5ab93e58bcba4425975ad00374da2bc/substrate/utils/wasm-builder/src/builder.rs#L212).

### CODE REVIEW

File: runtime/pendulum/Cargo.toml [hunk 1]
```diff
@@ -340,3 +342,5 @@ try-runtime = [
     "zenlink-protocol/try-runtime",
     "treasury-buyout-extension/try-runtime",
 ]
+
+metadata-hash = ["substrate-wasm-builder/metadata-hash"]
```
[the remark this answers is not in the document]
Comment (ebma) [reply runtime/pendulum/Cargo.toml:346]: Though in other parachains it's without the `?`. Maybe it's related to optional dependencies.  Probably safe to ignore.

**Review by ebma — APPROVED** [review]
> I am not sure if this change will break existing integration of our services (and external ones), since now the runtime is expecting an extra byte that will flag if the sender of the extrinsic wants to verify the metadata or not (see [flags](https://wiki.polkadot.network/docs/build-transaction-construction) here).

Let's just see how it goes on Foucoco. I think we don't need to explicitly define it but if we do, we can just define it in [pendulum.js](https://github.com/pendulum-chain/pendulum.js/blob/66d640334ecd7e012e21cb5392d30267bb47bb3e/packages/api/src/index.ts#L60) and upgrade the dependencies that are using pendulum.js already. Then it should be included automatically.

Once this is merged we also need to change our Gitlab release pipeline to include the new `--features metadata-hash` flag by default. The pipeline will compile the wasm runtimes and put it into the Github releases after all.

File: runtime/pendulum/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gianfra-t) [reply runtime/pendulum/Cargo.toml:346]: I saw it, but our `substrate-wasm-builder` dependency is not optional. Do you think we should make it optional too? In that case, adding the conditional would make sense.

[the remark this answers is not in the document]
Comment (ebma) [reply runtime/pendulum/Cargo.toml:346]: Mhh nah let's just keep it that way.

**Pull request merged**